internal/app/plan.go

```diff
@@ -135,47 +135,45 @@ func releaseWithHooks(cmd orderedCommand, storageBackend string, wg *sync.WaitGr
 		wg.Done()
 		<-sem
 	}()
-	if cmd.targetRelease == nil && !flags.destroy {
-		err := fmt.Errorf("nil target release")
-		errors <- err
-		log.Verbose(err.Error())
-		return
-	}
 	var annotations []string
-	for _, c := range cmd.beforeCommands {
-		if err := execOne(c.Command, cmd.targetRelease); err != nil {
-			errors <- err
+	if cmd.targetRelease != nil && !flags.destroy {
+		for _, c := range cmd.beforeCommands {
+			if err := execOne(c.Command, cmd.targetRelease); err != nil {
+				errors <- err
+				if key, err := c.getAnnotationKey(); err != nil {
```

Comment:
I know this wasn't introduced by this PR but I think this is wrong, when `getAnnotationKey` returns a non `nil` error, the `key` will be empty, so we should only append the annotation if the error is `nil` right? Or am I missing something?